docs/upgrades/v0.14-to-v1-upgrade/upgraded-examples/src/test/kotlin/examples/ErrorHandlingChanges.kt

```diff
@@ -0,0 +1,237 @@
+        assertEquals(SourceLocation(2, 15, 1), exception.location)
+        // Inspect properties from exception
+        assertEquals("UNRECOGNIZED", exception.get("TOKEN_NAME", String::class.java))
+    }
+
+    /**
+     * This example shows the error handling behavior that the [PartiQLPlanner] can proceed with most errors
+     * or warnings. However, it may still throw [PRuntimeException] in some circumstances
+     */
+    @Test
+    fun `PartiQLPlanner default error handling which throws exception`() {
+        val parser = PartiQLParser.standard()
+        val planner = PartiQLPlanner.standard()
+        val session = Session.empty()
+
+        // CREATE TABLE is not supported by planner currently
+        val parseResult = parser.parse("CREATE TABLE mytable")
+        val exception = assertThrows<PRuntimeException> {
+            try {
+                // Create are not supported in current Planner
+                val plan = planner.plan(parseResult.statements[0], session).plan
+            } catch (e: PRuntimeException) {
+                // Rethrow for validating the exception in test
+                throw e
+            }
+        }
+
+        assertEquals(PError.INTERNAL_ERROR, exception.error.code())
+        assertEquals(PErrorKind.SEMANTIC, exception.error.kind.code())
+        assertEquals(Severity.ERROR, exception.error.severity.code())
+    }
+
+    /**
+     * This example shows the error handling behavior that the [PartiQLPlanner] can proceed with most errors
+     * or warnings and will not throw.
+     */
+    @Test
+    fun `PartiQLPlanner default error handling which does not throw`() {
+        val parser = PartiQLParser.standard()
+        val planner = PartiQLPlanner.standard()
+        val session = Session.empty()
+
+        // Invalid expression adding number and string
+        val parseResult = parser.parse("SELECT 1 + 'string' FROM mytable")
+        assertDoesNotThrow {
+            val plan = planner.plan(parseResult.statements[0], session).plan
+        }
+    }
+
+    /**
+     * This example shows the error handling behavior that the [PartiQLPlanner] can proceed with most errors
+     * or warnings and will not throw. But we use error listener to trap errors or warnings from the planner.
+     */
+    @Test
+    fun `PartiQLPlanner error handling with custom listener`() {
+        // Register Error Listener
+        val listener = CustomPErrorListener()
+        val context = Context.of(listener)
+
+        val parser = PartiQLParser.standard()
+        val planner = PartiQLPlanner.standard()
+        val session = Session.empty()
+
+        // Invalid expression adding number and string
+        val parseResult = parser.parse("SELECT 1 + 'string' FROM mytable")
+        assertDoesNotThrow {
+            val plan = planner.plan(parseResult.statements[0], session, context).plan
+        }
+        assertEquals(2, listener.errorCollection.size)
```

Comment:
Should call out somewhere that there are two errors, not just the number and string addition.